### PR TITLE
Allow type cache query to fail

### PR DESCRIPTION
**Pull request by edhollandAL**
For version of PG below 9 the pg_type table has a different format this causes epgsql to fail to connect to AWS redshift as it is based on PG 8

### CODE REVIEW

File: src/epgsql.erl [hunk 1]
```diff
@@ -109,8 +109,14 @@ update_type_cache(C) ->
     Query = "SELECT typname, oid::int4, typarray::int4"
             " FROM pg_type"
             " WHERE typname = ANY($1::varchar[])",
-    {ok, _, TypeInfos} = equery(C, Query, [DynamicTypes]),
-    ok = gen_server:call(C, {update_type_cache, TypeInfos}).
+    case equery(C, Query, [DynamicTypes]) of
+        {ok, _, TypeInfos} ->
+            ok = gen_server:call(C, {update_type_cache, TypeInfos});
+        {error, _Error} ->
```
Comment (seriyps) [inline src/epgsql.erl]: Match with some concrete `_Error` would be better. For example, what if it's socket error or like that?

**Comment by edhollandAL** [comment]
Have updated to only silently ignore the specific error we encounter

**Comment by edhollandAL** [comment]
@seriyps: Have updated based on your comment. Are you happy with this now?

**Comment by seriyps** [comment]
yep, looks better.

**Comment by edhollandAL** [comment]
@davidw: are we ready to merge?

**Pull request merged**